### ISSUE TITLE
Multi-column constraint not handled correctly

### ISSUE THREAD

**Issue by SebastianEngel** [novoda/sqlite-provider#24]
I have the following relations table:

```sql
CREATE TABLE "image_likers" (
    _id INTEGER PRIMARY KEY AUTOINCREMENT,
    image_id INTEGER,
    user_id INTEGER,

    UNIQUE(image_id, user_id) ON CONFLICT REPLACE,
    FOREIGN KEY (image_id) REFERENCES "images" (_id),
    FOREIGN KEY (user_id) REFERENCES "users" (_id)
);
```

So what I would expect is that every relation, made of image_id and user_id is unique, not one column by itself. Trying that in an SQLite client works.

So having the following content would work and be ok:

1, 100, 200  // User 200 likes image 100
2, 101, 200  // User 200 likes image 101
3, 102, 200  // User 200 likes image 102

So like having the same user_id in multiple rows is fine, same for image_id. Just not the same combinations.

Adding another row like this should not add another one but replace row 1:
image_id = 100, user_id = 200

Using an SQLite client, all is fine with that. But the SQLiteProvider only evaluates one unique constraint, so for one column. The problem seems to be in InsertHelper.insert(...).

```java
final String firstConstrain = dbHelper.getFirstConstrain(table, insertValues);
```

This line returns only the "image_id" column as unique constraint and then tries an update on that:

```sql
UPDATE image_likers SET image_id=?,user_id=? WHERE image_id=?
```

As a result, whenever it finds an existing image_id, it updates the row, which means that all of the images have at most 1 liker.
Instead it should only update the row if it has the same image_id - user_id combination.

Might that be a bug?

**Comment by xrigau** [comment]
Fixed in #47 which will be released soon

**Comment by ronocod** [comment]
Fixed in [v1.0.6](https://github.com/novoda/sqlite-provider/releases/tag/1.0.6)

**Comment by xrigau** [comment]
1.0.6 was released incorrectly - we'll fix this next week
